kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/net/EndpointRegistry.java

```diff
@@ -618,6 +624,35 @@ public CompletionStage<Void> shutdown() {
         return allOfStage(registeredVirtualClusters.keySet().stream().map(this::deregisterVirtualCluster));
     }
 
+    /**
+     * Selects a bootstrap server from the list using a simple round-robin approach.
+     * This distributes load across multiple bootstrap servers and provides better resilience when one server is down.
+     *
+     * @param bootstrapServers the list of bootstrap servers
+     * @return the selected bootstrap server
+     */
+    static HostPort selectBootstrapServer(List<HostPort> bootstrapServers) {
+        return selectBootstrapServer(bootstrapServers, roundRobinCounter.getAndIncrement());
+    }
+
+    /**
+     * Selects a bootstrap server from the list using a simple round-robin approach for testing.
+     * This version accepts a counter value for deterministic testing.
+     *
+     * @param bootstrapServers the list of bootstrap servers
+     * @param counter the counter value to use for selection
+     * @return the selected bootstrap server
+     */
+    @VisibleForTesting
+    static HostPort selectBootstrapServer(List<HostPort> bootstrapServers, long counter) {
+        if (bootstrapServers == null || bootstrapServers.isEmpty()) {
+            throw new IllegalArgumentException("Bootstrap servers list cannot be null or empty");
+        }
+
+        int index = (int) (counter % bootstrapServers.size());
```

Comment:
I know the likelihood of on AtomicLong cycling round Long.MIN is quite remote, but I'd still wrap in Math.abs() to avoid the negative index possibility.